### PR TITLE
feat: implemented #202

### LINKED ISSUE #202: sshnp: add -u option to specify username to use if sshnpd is not sharing it
### Is your feature request related to a problem? Please describe.

from @cconstab 
> sshnp shoukd have a -u option to specify the username if the sshnpd was not sharing it.
> This is a bug/oversight in sshnp right now.

### Describe the solution you'd like

* add an optional `-u` option in SSHNP.createArgParser with full option name `remote-user-name`, description something like, "user name to use in the ssh session on the remote host"
* in SSHNP, change `remoteUserName` from `late final String remoteUsername` to `String? remoteUserName`
* add optional `remoteUserName` parameter to SSHNP constructor to allow remoteUserName to be supplied, and supply the parsed 
* in SSHNP.init - do not call `fetchRemoteUserName` if a remoteUserName has already been supplied

### Describe alternatives you've considered

_No response_

### Additional context

Either wait for https://github.com/atsign-foundation/sshnoports/pull/201 to be merged to trunk, or make a new branch based on that PR's branch

### PR DESCRIPTION
**Pull request by purnimavenkatasubbu**
**- What I did**
Made changes as per comments in ticket #202 

**- How I did it**
Made changes as per comments in ticket #202 

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: lib/sshnp.dart [hunk 1]
```diff
@@ -2,23 +2,21 @@
...
-import 'package:version/version.dart';
-
+import 'package:sshnoports/cleanup_sshnp.dart';
 // local packages
 import 'package:sshnoports/service_factories.dart';
 import 'package:sshnoports/sshnp_utils.dart';
-import 'package:sshnoports/cleanup_sshnp.dart';
 import 'package:sshnoports/version.dart';
+import 'package:uuid/uuid.dart';
+import 'package:version/version.dart';
```
Comment (gkc) [inline lib/sshnp.dart]: Why have all the import statements moved around?

File: lib/sshnp.dart [hunk 2]
```diff
@@ -87,7 +83,7 @@ class SSHNP {
 
   /// The username to use on the remote host in the ssh session. Is fetched
   /// from the sshnpd by [fetchRemoteUserName] during [init]
-  late final String remoteUsername;
+  String? remoteUsername;
```
Comment (gkc) [inline lib/sshnp.dart:89]: When you change the code, the documentation _must_ also be updated

File: lib/sshnp.dart [hunk 3]
```diff
@@ -299,13 +302,13 @@ class SSHNP {
     }
   }
 
-
   /// Look up the user name ... we expect a key to have been shared with us by
   /// sshnpd. Let's say we are @human running sshnp, and @daemon is running
   /// sshnpd, then we expect a key to have been shared whose ID is
   /// @human:username.device.sshnp@daemon
   Future<void> fetchRemoteUserName() async {
```
Comment (gkc) [inline lib/sshnp.dart:311]: Update the documentation here also to explain that this is not called when a remoteUserName is supplied at construction time

File: lib/sshnp.dart [hunk 4]
```diff
@@ -477,6 +480,11 @@ class SSHNP {
         throw ('\nUnable to determine your username: please set environment variable\n\n');
       }
 
+      String? remoteUsername;
+      if (results.wasParsed('remote-user-name')) {
+        remoteUsername = results['remote-user-name'];
+      }
```
Comment (gkc) [inline lib/sshnp.dart]: You don't need this code

File: lib/sshnp.dart [hunk 5]
```diff
@@ -545,6 +553,7 @@ class SSHNP {
         localSshOptions: results['local-ssh-options'] ?? [],
         rsa: results['rsa'],
         sendSshPublicKey: sendSshPublicKey,
+        remoteUsername: remoteUsername,
```
Comment (gkc) [inline lib/sshnp.dart]: change to `remoteUsername: results['remote-user-name']`,

**Review by gkc — CHANGES_REQUESTED** [review]

File: lib/sshnp.dart (hunk 1 above)
Comment (purnimavenkatasubbu) [reply lib/sshnp.dart]: Fixed

File: lib/sshnp.dart (hunk 2 above)
Comment (purnimavenkatasubbu) [reply lib/sshnp.dart:89]: Updated the documentation

File: lib/sshnp.dart (hunk 3 above)
Comment (purnimavenkatasubbu) [reply lib/sshnp.dart:311]: Updated!

File: lib/sshnp.dart (hunk 5 above)
Comment (purnimavenkatasubbu) [reply lib/sshnp.dart]: Updated

File: lib/sshnp.dart [hunk 6]
```diff
@@ -85,9 +83,10 @@ class SSHNP {
   @visibleForTesting
   late final String clientAtSign;
 
-  /// The username to use on the remote host in the ssh session. Is fetched
-  /// from the sshnpd by [fetchRemoteUserName] during [init]
-  late final String remoteUsername;
+  /// The username to use on the remote host in the ssh session. Either passed
+  /// through class constructor or  Is fetched from the sshnpd
```
Comment (gkc) [inline lib/sshnp.dart]: Please correct capitalization and spacing

File: lib/sshnp.dart [hunk 7]
```diff
@@ -299,13 +305,14 @@ class SSHNP {
     }
   }
 
-
   /// Look up the user name ... we expect a key to have been shared with us by
   /// sshnpd. Let's say we are @human running sshnp, and @daemon is running
   /// sshnpd, then we expect a key to have been shared whose ID is
   /// @human:username.device.sshnp@daemon
+  /// fetches remote user name if not provided through class constructor
```
Comment (gkc) [inline lib/sshnp.dart]: Replace with "Is not called if remoteUserName was set via constructor"

**Review by gkc — CHANGES_REQUESTED** [review]

File: lib/sshnp.dart (hunk 7 above)
Comment (purnimavenkatasubbu) [reply lib/sshnp.dart]: Done

File: lib/sshnp.dart (hunk 6 above)
Comment (purnimavenkatasubbu) [reply lib/sshnp.dart]: Done

**Review by gkc — APPROVED** [review]
LGTM thank you @purnimavenkatasubbu 

@cconstab can you review this also please?

**Review by gkc — APPROVED** [review]

**Pull request merged**